services/src/handlers/workflows.rs

```diff
@@ -240,19 +257,299 @@ async fn get_workflow_provenance<C: Context>(
+/// Returns the id of the created dataset and upload
+///
+/// # Example
+///
+/// ```text
+/// POST /datasetFromWorkflow/{workflow_id}
+/// Authorization: Bearer fc9b5dc2-a1eb-400f-aeed-a7845d9935c9
+/// Content-Type: application/json
+///
+/// {
+///     "name": "foo",
+///     "description": null,
+///     "query": {
+///         "spatial_bounds": {
+///             "upperLeftCoordinate": {
+///                 "x": -10.0,
+///                 "y": 80.0
+///             },
+///             "lowerRightCoordinate": {
+///                 "x": 50.0,
+///                 "y": 20.0
+///             }
+///         },
+///         "time_interval": {
+///             "start": 1388534400000,
+///             "end": 1388534401000
+///         },
+///         "spatial_resolution": {
+///             "x": 0.1,
+///             "y": 0.1
+///         }
+///     }
+/// }
+///
+/// ```text
+/// {
+///   "upload": "3086f494-d5a4-4b51-a14b-3b29f8bf7bb0",
+///   "dataset": {
+///     "type": "internal",
+///     "datasetId": "94230f0b-4e8a-4cba-9adc-3ace837fe5d4"
+///   }
+/// }
+/// ```
+pub(crate) fn dataset_from_workflow_handler<C: Context>(
+    ctx: C,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = warp::Rejection> + Clone {
+    // TODO: support vector datasets
+
+    warp::path!("datasetFromWorkflow" / Uuid)
+        .map(WorkflowId)
+        .and(warp::post())
+        .and(authenticate(ctx.clone()))
+        .and(warp::any().map(move || ctx.clone()))
+        .and(warp::body::json())
+        .and_then(dataset_from_workflow)
+}
+
+// TODO: move into handler once async closures are available?
+async fn dataset_from_workflow<C: Context>(
+    workflow_id: WorkflowId,
+    session: C::Session,
+    ctx: C,
+    info: RasterDatasetFromWorkflow,
+) -> Result<impl warp::Reply, warp::Rejection> {
+    // TODO: support datasets with multiple time steps
+
+    let workflow = ctx.workflow_registry_ref().await.load(&workflow_id).await?;
+
+    let operator = workflow.operator.get_raster().context(error::Operator)?;
+
+    let execution_context = ctx.execution_context(session.clone())?;
+    let initialized = operator
+        .clone()
+        .initialize(&execution_context)
+        .await
+        .context(error::Operator)?;
+
+    let result_descriptor = initialized.result_descriptor();
+
+    let processor = initialized.query_processor().context(error::Operator)?;
+
+    // put the created data into a new upload
+    let upload = UploadId::new();
+    let upload_path = upload.root_path()?;
+    fs::create_dir_all(&upload_path).await.context(error::Io)?;
+    let file_path = upload_path.join("raster.tiff");
+
+    let query_rect = info.query;
+    let query_ctx = ctx.query_context()?;
+    let no_data_value = result_descriptor.no_data_value;
+    let request_spatial_ref =
+        Option::<SpatialReference>::from(result_descriptor.spatial_reference).unwrap(); // TODO
+    let tile_limit = None; // TODO
```

Comment:
Please state what is `todo`

Reply:
09ce2d78b2cf0251d2c21f7396880121fb0286c6